include/graphit/midend/mir_emitter.h

```diff
@@ -168,6 +168,12 @@ namespace graphit {
 
         void addElementType(mir::ElementType::Ptr);
 
+        std::vector<std::string> emitFunctorArgs(std::vector<fir::Expr::Ptr> functorArgs);
```

Comment:
Let's not use strings here. It will not work if the arguments aren't variables but other expressions.

Reply:
Ah yeah. For now we are assuming it can only be Var or Literal expressions.

Reply:
but i agree it should be extended to anytime of expressions

Reply:
Either way is fine for me. A lot of things might have to be changed to make it work for any expression, which might be non-trivial.